### ISSUE TITLE
Merge ember-cli-create into ember-cli / Configuration Authoring Format

### ISSUE THREAD

**Issue by gossi** [ember-cli/ember-cli#8343]
At Emberfest 2018 I demoed `ember-cli-create`.

You can find the video here: https://www.youtube.com/watch?v=mZD0aaf3IoA

Of course it is my pleasure to at some point merge it back into ember-cli itself. Also based on the feedback I received it is admired very hearth welcoming. 

I thought I put together what I see needed to nail the merge and truly make benefits of the concepts that come with it. Most of them I outlined in my blog post: [Thoughts on Improving Developer Ergonomics for emberjs](https://gos.si/blog/thoughts-on-improving-developer-ergonomics-for-emberjs).

First make yourself comfortable with the lengthy but great [embroider spec](https://github.com/embroider-build/embroider/blob/master/SPEC.md) by @ef4. He is talking about the the _publication_ format for packages while I describe the _authoring_ format in my blog post. One problem described from different angles and with ideas for solution. I think a new authoring format can be worked out (given it will receive a different filename) and people can opt-in to the new format by creating the new file (the old one will then be ignored). Despite working out a structure for a new authoring format, it's also to define what blueprints and presets are (hint: I see presets as a way to parametrize blueprints), these configuration in presets need a slot in the authoring format to be docked into.

Second is to run more research on what categories the `ember-cli-create` wizard will take. That can happen with the current addon (I have several ideas) for the time being.

A summary on a very high level:

- [ ] Work out an authoring format
- [ ] Define blueprints and presets
- [ ] Implement new authoring format
- [ ] Research the UI wizard for `ember-cli-create`

**Comment by stefanpenner** [comment]
Although I do like `ember-cli-create`, wizard style boot-strap tools are tricky to maintain and support. 

The original designs of ember-cli specifically took a non wizard approach (but with `-t` for custom templates/blueprints as an override) to avoid the "large matrix of possibilities" problem wizard approaches create. Specifically, the concerns were:

* ensuring the full possible matrix of generated apps work, and don't break (without exploding test suite times)
* ensuring the "base-app" is a reasonable default to build an ecosystem on.

I believe with proper design and thought the above concerns can be dealt with, but that design must occur.

**Comment by gossi** [comment]
Thanks @stefanpenner - there absolutely two parts invovled here, thanks for pointing out those directions. Consider this issue as a meta/quest thing then.

PS. I mainly put this up mainly, because I now can link to this issue when people ask me =)

**Comment by kellyselden** [comment]
We chatted about this on a call a few weeks back. We all agreed ember-cli-create was a great project, and we would be very interested in using its wizard style for our existing options (yarn, welcome screen, etc). When we started talking about bundling third party stuff like less, composable helpers, etc, veryifying they always work in different combos, avoiding one person bus factors, the maintenence burden seems really big and I'm not convinced yet we should do it. I currently think it existing in user-land is a good place for it.

If this does become an RFC, I'm curious:
 * has vue-cli had any issues supporting this?
 * is it worth stretching our maintence time across all configs instead of making sure a base app works great?
 * how are projects like ember-cli-update going to adapt when no two projects start out with the same base app
 * do we risk bundling ecosystem addons, or do we require they hand over their keys to us?
